packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/chat.tests.js

```diff
@@ -1091,10 +1089,397 @@ QUnit.module('Chat', () => {
+            this.clock.tick(timeout);
+
+            assert.strictEqual(this.getDayHeaders().length, 2, 'day header was removed');
+            assert.strictEqual(this.getDayHeaders().eq(0).text(), '10/17/2021', 'day header content is correct');
+            assert.strictEqual(this.getDayHeaders().eq(1).text(), '10/27/2021', 'day header content is correct');
+        });
+
+        QUnit.test('day header should be removed after the last message is deleted', function(assert) {
+            const messages = [{
+                id: 1,
+                text: 'message_1',
+                timestamp: new Date('2021/10/17'),
+                author: userFirst
+            }];
+
+            const timeout = 100;
+
+            const store = new CustomStore({
+                key: 'id',
+                load: function() {
+                    const d = $.Deferred();
+                    setTimeout(function() {
+                        d.resolve([...messages]);
+                    }, timeout);
+                    return d.promise();
+                },
+            });
+
+            this.reinit({
+                dataSource: store,
+                user: { id: userFirst.id },
+                reloadOnChange: false,
+            });
+
+            this.clock.tick(timeout);
+
+            assert.strictEqual(this.getDayHeaders().length, 1, 'day header should be present');
+
+            store.push([{ type: 'remove', key: 1 }]);
+            this.clock.tick(timeout);
+
+            assert.strictEqual(this.getDayHeaders().length, 0, 'day header was removed');
+        });
+
+        QUnit.test('emptyview should be rendered after the last message is deleted from the store', function(assert) {
+            const messages = [{
+                id: 1,
+                text: 'message_1',
+                timestamp: new Date('2021/10/17'),
+                author: userFirst
+            }];
+
+            const timeout = 100;
+
+            const store = new CustomStore({
+                key: 'id',
+                load: function() {
+                    const d = $.Deferred();
+                    setTimeout(function() {
+                        d.resolve([...messages]);
+                    }, timeout);
+                    return d.promise();
+                },
+            });
+
+            this.reinit({
+                dataSource: store,
+                user: { id: userFirst.id },
+                reloadOnChange: false,
+            });
+
+            this.clock.tick(timeout);
+
+            assert.strictEqual(this.getEmptyView().length, 0, 'empty view is not rendered');
+
+            store.push([{ type: 'remove', key: 1 }]);
+            this.clock.tick(timeout * 2);
+
+            assert.strictEqual(this.getEmptyView().length, 1, 'empty view is rendered');
+        });
+
+        QUnit.test('emptyview should be rendered after each removal and removed after each addition from the store', function(assert) {
```

Comment:
I didn't get what u're testing here. The test name says we should add/remove emptyView after reach add/remove operation.. But you assert once after all operations, so it looks the same as the ["emptyview should be rendered after the last message is deleted from the store" test](https://github.com/DevExpress/DevExtreme/pull/28304/files#diff-c28d8d0fcc791fc81e519f30bf992c6eb3607f134382d592b9518688d76a640eR1390).

Reply:
Thanks. Scenario was simplified: https://github.com/DevExpress/DevExtreme/pull/28304/commits/f7b68176e3e980d976fa7a3b8f77a8fa36812a9b